tests/test_spectrum_processing.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+from matchms import Spectrum
+from ms2query.spectrum_processing import require_peaks_below_mz, \
+    spectrum_processing_minimal, spectrum_processing_s2v
+
+
+
+def test_require_peaks_below_mz_no_params():
+    """Test default"""
+    mz = np.array([110, 220, 330, 440], dtype="float")
+    intensities = np.array([0, 1, 10, 100], dtype="float")
+    spectrum_in = Spectrum(mz=mz, intensities=intensities)
+
+    spectrum = require_peaks_below_mz(spectrum_in)
+
+    assert spectrum is None, "Expected None because the number of peaks (4) is less than the default threshold (10)."
+
+
+def test_require_peaks_below_mz_required_4():
+    """Test with adjustment of n_required."""
+    mz = np.array([110, 220, 330, 440], dtype="float")
+    intensities = np.array([0, 1, 10, 100], dtype="float")
+    spectrum_in = Spectrum(mz=mz, intensities=intensities)
+
+    spectrum = require_peaks_below_mz(spectrum_in, n_required=4)
+
+    assert spectrum == spectrum_in, "Expected the spectrum to qualify because the number of peaks (4) is equal to the" \
+                                "required number (4)."
+
+
+def test_require_peaks_below_mz_required_4_below_max_mz():
+    """Test with adjustment of n_required but with lower max_mz."""
+    mz = np.array([110, 220, 330, 440], dtype="float")
+    intensities = np.array([0, 1, 10, 100], dtype="float")
+    spectrum_in = Spectrum(mz=mz, intensities=intensities)
+
+    spectrum = require_peaks_below_mz(spectrum_in, n_required=4, max_mz=400)
+
+    assert spectrum is None, \
+        "Expected None since peaks with mz<=max_mz is less than n_required."
+
+
+def test_require_peaks_below_empty_spectrum():
```

Comment:
Nice thorough testing of test_require_peaks_below_mz

Reply:
Thanks.